subiquity/server/server.py

```diff
@@ -723,7 +727,66 @@ async def start_site(self, runner: web.AppRunner):
     def base_relative(self, path):
         return os.path.join(self.base_model.root, path)
 
-    def load_cloud_config(self):
+    @with_context(name="extract_autoinstall")
+    async def _extract_autoinstall_from_cloud_config(
+        self,
+        *,
+        cloud_cfg: dict[str, Any],
+        context: Context,
+    ) -> dict[str, Any]:
+        """Extract autoinstall passed via cloud config."""
+
+        # Not really is-install-context but set to force event reporting
+        context.set("is-install-context", True)
+        context.enter()  # publish start event
+
+        if "autoinstall" not in cloud_cfg:
```

Comment:
Well, part of the problem is that people aren't supplying an autoinstall key, but supplying autoinstall directives directly to cloud-init  (or could be doing both, but I'm guessing that's not common).  So I think we need to revisit this part.  retrieving `cloud_cfg["autoinstall"]` could be after `validate_cloud_init_schema()` in the current setup.

Reply:
That makes sense to me. This probably flows better with the throws-on-failure logic.